orm/internal/testpb/bank_query.proto

```diff
@@ -3,145 +3,111 @@ syntax = "proto3";
 package testpb;
 
 import "cosmos/base/query/v1beta1/pagination.proto";
+import "google/api/annotations.proto";
 import "testpb/bank.proto";
 
 // BankQueryService queries the state of the tables specified by testpb/bank.proto.
 service BankQueryService {
-  // Get queries the Balance table by its primary key.
-  rpc GetBalance(GetBalanceRequest) returns (GetBalanceResponse) {}
-  // ListBalance queries the Balance table using prefix and range queries against defined indexes.
-  rpc ListBalance(ListBalanceRequest) returns (ListBalanceResponse) {}
-  // Get queries the Supply table by its primary key.
-  rpc GetSupply(GetSupplyRequest) returns (GetSupplyResponse) {}
-  // ListSupply queries the Supply table using prefix and range queries against defined indexes.
-  rpc ListSupply(ListSupplyRequest) returns (ListSupplyResponse) {}
+  // Balance queries the Balance table by its primary key.
+  rpc Balance(BalanceRequest) returns (BalanceResponse) {
+    option (google.api.http) = {
+      get: "/testpb/balances/{address}/{denom}"
+    };
+  }
+  // Balances queries the Balance table using the primary key index.
+  rpc Balances(BalancesRequest) returns (BalancesResponse) {
+    option (google.api.http) = {
+      get: "/testpb/balances"
+    };
+  }
+  // BalancesByDenom queries the Balance table using the primary key index.
+  rpc BalancesByDenom(BalancesByDenomRequest) returns (BalancesByDenomResponse) {
+    option (google.api.http) = {
+      get: "/testpb/balances.denom"
```

Comment:
Is this conventional? Can this be more user friendly? e.g. `balances-by-denom`

Reply:
Also worth noting the sdk uses `_` but as far as I understand `-` is more conventional.

Reply:
No it's not conventional. Initially I was thinking an [HTTP matrix parameter](https://www.w3.org/DesignIssues/MatrixURIs.html) would be nice (`balances;index=denom`) but that doesn't play so nicely with grpc gateway.

The thing that bothers me about `balances-by-denom` is that it isn't actually a domain type but rather an index, but maybe that's okay. I also considered `balances/by-denom` but that conflicts with the actual balances GET query sort of.

Reply:
Matrix parameters look interesting but also not yet widely adopted, at least as far as I've seen.

Yea, might be nice to use the actual domain type with an index. A nice thing about `balances-by-denom` is that the CLI command is `balance-by-denom` and the name of the query is `BalancesByDenom`.

We had inconsistent endpoint names in Regen Ledger, which we resolved by adding alternatives, each endpoint reflects the name of the query method and then we added more conventional endpoints. We might have chosen one or the other but wanted to limit breaking changes so we chose to support both in the meantime. For example:

```
    option (google.api.http) = {
      get : "/regen/ecocredit/v1/projects-by-class/{class_id}"
      additional_bindings : [
        {get : "/regen/ecocredit/v1/projects/class/{class_id}"},
        {get : "/regen/ecocredit/v1/classes/{class_id}/projects"}
      ]
    };
```